Mathlib/RingTheory/TensorProduct.lean

```diff
@@ -393,23 +435,66 @@ end ext
 
 end Semiring
 
-section Ring
-variable [CommRing R]
-variable [Ring A] [Algebra R A]
-variable [Ring B] [Algebra R B]
+section AddCommGroupWithOne
+variable [CommSemiring R]
+variable [AddCommGroupWithOne A] [Module R A]
+variable [AddCommGroupWithOne B] [Module R B]
 
-instance instRing : Ring (A ⊗[R] B) where
-  zsmul := SubNegMonoid.zsmul
-  zsmul_zero' := SubNegMonoid.zsmul_zero'
-  zsmul_succ' := SubNegMonoid.zsmul_succ'
-  zsmul_neg' := SubNegMonoid.zsmul_neg'
+instance instAddCommGroupWithOne : AddCommGroupWithOne (A ⊗[R] B) where
+  toAddCommGroup := TensorProduct.addCommGroup
+  __ := instAddCommMonoidWithOne
   intCast z := z ⊗ₜ (1 : B)
   intCast_ofNat n := by simp [natCast_def]
   intCast_negSucc n := by simp [natCast_def, add_tmul, neg_tmul, one_def]
-  add_left_neg := add_left_neg
 
 theorem intCast_def (z : ℤ) : (z : A ⊗[R] B) = (z : A) ⊗ₜ (1 : B) := rfl
 
+end AddCommGroupWithOne
```

Comment:
As with `natCast_def`, can you please provide `intCast_right`?

Reply:
#7391